pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -550,7 +551,7 @@ func getConditionCheckName(taskRunStatus map[string]*v1beta1.PipelineRunTaskRunS
 			}
 		}
 	}
-	return names.SimpleNameGenerator.RestrictLengthWithRandomSuffix(fmt.Sprintf("%s-%s", trName, conditionRegisterName))
```

Comment:
Is the names package still used anywhere?

Reply:
Yes, it's still used to generate step names, script filenames, internal volumes mount-point names.

I'm not sure whether it's actually needed in all of those places, but I wanted to focus on k8s resources first in this PR - which is where we have experience issues. Which has the advantage of making this PR easier to approach and review too :)